segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/rolling/RollingStorage.java

```diff
@@ -501,6 +511,36 @@ private void sealActiveChunk(RollingSegmentHandle handle) throws StreamSegmentEx
         }
     }
 
+    private void unsealLastChunkIfNecessary(RollingSegmentHandle handle) throws StreamSegmentException {
```

Comment:
This is only called from one location, and it already has the chunk that it wants to unseal. Perhaps it should pass it directly that way we aren't vulnerable to bugs related to somehow 'last' changing.

Reply:
I get what you're saying, but that method makes use of other methods on the `RollingSegmentHandle`. I'd have to pass both the handle and the `last` reference which I obtained just prior. IMO, that would open the door for even more bugs, in case we pass the "wrong" reference to the last chunk.